dali/dfu/dfuwu.cpp

```diff
@@ -233,6 +245,70 @@ StringBuffer &encodeDFUsortfield(DFUsortfield fmt,StringBuffer &str,bool incmodi
     return str.append(DFUsortfields[i].str);
 }
 
+bool DfuParseLogicalPath(const IPropertyTree *directories, const char * pLogicalPath, const char* groupName,
+                StringBuffer &folder, StringBuffer &title, StringBuffer &defaultFolder, StringBuffer &defaultReplicateFolder)
+{
+    if(!pLogicalPath || !*pLogicalPath)
+        return false;
+
+    folder.clear();
+    title.clear();
+
+    defaultFolder.clear();
+    defaultReplicateFolder.clear();
+
+    if(groupName != NULL && *groupName != '\0')
+    {
+#ifdef _CONTAINERIZED
+        Owned<IPropertyTree> plane = getStoragePlane(groupName);
+        if (plane)
+            defaultFolder.append(plane->queryProp("@prefix"));
+#else
+        StringBuffer basedir;
+        GroupType groupType;
+        Owned<IGroup> group = queryNamedGroupStore().lookup(groupName, basedir, groupType);
+        if (group)
+        {
+            if (directories)
+            {
+                switch (groupType)
+                {
+                case grp_roxie:
+                    getConfigurationDirectory(directories, "data", "roxie", nullptr, defaultFolder);
+                    getConfigurationDirectory(directories, "data2", "roxie", nullptr, defaultReplicateFolder);
+                    // MORE - should extend to systems with higher redundancy
+                    break;
+                case grp_hthor:
+                    getConfigurationDirectory(directories, "data", "eclagent", nullptr, defaultFolder);
+                    break;
+                case grp_thor:
+                default:
+                    getConfigurationDirectory(directories, "data", "thor", nullptr, defaultFolder);
+                    getConfigurationDirectory(directories, "mirror", "thor", nullptr, defaultReplicateFolder);
+                }
+            }
+        }
+        else
+        {
+            // Ignore at least for now for backward compatability.
+        }
+#endif
+    }
+
+    getLFNDirectoryUsingBaseDir(folder, pLogicalPath, defaultFolder.str());
+
+    const char *n = pLogicalPath;
+    const char* p;
+    do {
+        p = strstr(n,"::");
+        if(p)
+            n = p+2;
+    } while(p);
+    title.append(n);
```

Comment:
for slightly loss of efficiency may be better to use standard lfn parsing code like:
```
CDfsLogicalFileName lfn(pLogicalPath);
lfn.getTail(title);
```